solidity/contracts/ETHStaking.sol

```diff
@@ -0,0 +1,65 @@
+/**
+▓▓▌ ▓▓ ▐▓▓ ▓▓▓▓▓▓▓▓▓▓▌▐▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓ ▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓ ▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▄
+▓▓▓▓▓▓▓▓▓▓ ▓▓▓▓▓▓▓▓▓▓▌▐▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓ ▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓ ▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓
+  ▓▓▓▓▓▓    ▓▓▓▓▓▓▓▀    ▐▓▓▓▓▓▓    ▐▓▓▓▓▓   ▓▓▓▓▓▓     ▓▓▓▓▓   ▐▓▓▓▓▓▌   ▐▓▓▓▓▓▓
+  ▓▓▓▓▓▓▄▄▓▓▓▓▓▓▓▀      ▐▓▓▓▓▓▓▄▄▄▄         ▓▓▓▓▓▓▄▄▄▄         ▐▓▓▓▓▓▌   ▐▓▓▓▓▓▓
+  ▓▓▓▓▓▓▓▓▓▓▓▓▓▀        ▐▓▓▓▓▓▓▓▓▓▓         ▓▓▓▓▓▓▓▓▓▓▌        ▐▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓
+  ▓▓▓▓▓▓▀▀▓▓▓▓▓▓▄       ▐▓▓▓▓▓▓▀▀▀▀         ▓▓▓▓▓▓▀▀▀▀         ▐▓▓▓▓▓▓▓▓▓▓▓▓▓▓▀
+  ▓▓▓▓▓▓   ▀▓▓▓▓▓▓▄     ▐▓▓▓▓▓▓     ▓▓▓▓▓   ▓▓▓▓▓▓     ▓▓▓▓▓   ▐▓▓▓▓▓▌
+▓▓▓▓▓▓▓▓▓▓ █▓▓▓▓▓▓▓▓▓ ▐▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓ ▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓  ▓▓▓▓▓▓▓▓▓▓
+▓▓▓▓▓▓▓▓▓▓ ▓▓▓▓▓▓▓▓▓▓ ▐▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓ ▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓  ▓▓▓▓▓▓▓▓▓▓
+
+                           Trust math, not hardware.
+*/
+
+pragma solidity 0.5.17;
+
+import "@keep-network/keep-core/contracts/Authorizations.sol";
+import "@keep-network/keep-core/contracts/StakeDelegatable.sol";
+import "@keep-network/keep-core/contracts/KeepRegistry.sol";
+
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+
+
+/// @title ETH Staking
+/// @notice A staking contract for ETH staking. An owner of the ETH can delegate
+/// ETH as a stake to an operator. The value of ETH the owner is willing to stake
+/// should be deposited in `ETHBonding` contract for the given operator.
+contract ETHStaking is Authorizations, StakeDelegatable {
+    constructor(KeepRegistry keepRegistry)
+        public
+        Authorizations(keepRegistry)
+    {}
+
+    event Staked(
+        address owner,
+        address indexed operator,
+        address indexed beneficiary,
+        address indexed authorizer
+    );
+
+    /// @notice Registers stake details. The function is used to register
+    /// addresses of operator, beneficiary and authorizer for a stake from the
+    /// caller.
+    /// @param operator Address of the operator.
+    /// @param beneficiary Address of the beneficiary.
+    /// @param authorizer Address of the authorizer.
+    function stake(
+        address operator,
+        address payable beneficiary,
+        address authorizer
+    ) external payable {
+        // TODO: Do we need to verify the caller in any way?
+        address _from = msg.sender;
+
+        require(
+            operators[operator].owner == address(0),
+            "Operator already in use"
+        );
+
+        operators[operator] = Operator(0, _from, beneficiary, authorizer);
+        ownerOperators[_from].push(operator);
+
+        emit Staked(_from, operator, beneficiary, authorizer);
+    }
```

Comment:
Worth noting, we'll need to have an initialization period here similar to how we do it for KEEP staking. Maybe you already have it in plans, just calling it out.

Reply:
Good point. We may need to handle it in a separate PR as this may require updates to FullyBackedSortitionPool.

Reply:
1e86e1a197b477af6ff7ebc09477afadc5dd8817